### DISCUSSION TITLE
`DatabaseConnection` clone creates a new pool?

### DISCUSSION THREAD

**Discussion by ziimakc** [SeaQL/sea-orm#2198] [Q&A] [answered]
I need to pass a connection to a multiple services and would like to know if cloning database connection creates a new pool? For example if 1 poll creates 5 connections and I will clone poll 100 times does it mean I will create 500 active db connections? If so, is there an easy way to use a shared connection pool without lifetimes?

```
let mut opt = ConnectOptions::new("protocol://username:password@host/database");
opt.max_connections(100)
    .min_connections(5)
    .connect_timeout(Duration::from_secs(8))
    .acquire_timeout(Duration::from_secs(8))
    .idle_timeout(Duration::from_secs(8))
    .max_lifetime(Duration::from_secs(8))
    .sqlx_logging(true)
    .sqlx_logging_level(log::LevelFilter::Info)
    .set_schema_search_path("my_schema"); // Setting default PostgreSQL schema

let db = Database::connect(opt).await?;
db.clone() // does it?
```

**Comment by billy1624** [comment, marked as answer]
Hey @ziimakc, sorry I make a mistake.

`sea_orm::DatabaseConnection` is based on [`sqlx::Pool`](https://docs.rs/sqlx/latest/sqlx/struct.Pool.html). 

As stated in the SQLx documentation.

> Cloning Pool is cheap as it is simply a reference-counted handle to the inner pool state. When the last remaining handle to the pool is dropped, the connections owned by the pool are immediately closed (also by dropping). 

So, the connection pool above will keep at least 5 active connections at all time and at most 100 connections. You can feel free to clone `sea_orm::DatabaseConnection`.
